engine/src/flutter/shell/platform/android/io/flutter/plugin/view/SensitiveContentPlugin.java

```diff
@@ -0,0 +1,116 @@
+// found in the LICENSE file.
+
+package io.flutter.plugin.view;
+
+import static io.flutter.Build.API_LEVELS;
+
+import android.app.Activity;
+import android.os.Build;
+import android.view.View;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.systemchannels.SensitiveContentChannel;
+import io.flutter.plugin.common.MethodChannel;
+
+/**
+ * {@link SensitiveContentPlugin} is the implementation of all functionality needed to set and
+ * retrieve the content sensitivity level of a native Flutter Android {@code View}.
+ *
+ * <p>This plugin handles requests for getting and setting content sensitivity sent by the {@link
+ * io.flutter.embedding.engine.systemchannels.SensitiveContentChannel}.
+ */
+public class SensitiveContentPlugin
+    implements SensitiveContentChannel.SensitiveContentMethodHandler {
+
+  private final int mFlutterViewId;
+  private final Activity mFlutterActivity;
+  private final SensitiveContentChannel mSensitiveContentChannel;
+
+  public SensitiveContentPlugin(
+      @NonNull int flutterViewId,
+      @NonNull Activity activity,
+      @NonNull SensitiveContentChannel sensitiveContentChannel) {
+    mFlutterActivity = activity;
+    mFlutterViewId = flutterViewId;
+    mSensitiveContentChannel = sensitiveContentChannel;
+
+    mSensitiveContentChannel.setSensitiveContentMethodHandler(this);
+  }
+
+  /**
+   * Sets content sensitivity level of the Android {@code View} associated with this plugin's {@code
+   * mFlutterViewId} to the level specified by {@requestedContentSensitivity}.
+   */
+  @Override
+  public void setContentSensitivity(
+      @NonNull int requestedContentSensitivity, @NonNull MethodChannel.Result result) {
+    if (!isSupported()) {
+      // This feature is only available on >= API 35.
+      result.error(
+          "error",
+          "Setting content sensitivity with value "
+              + requestedContentSensitivity
+              + " failed because this feature is only available on Android API >= 35.",
+          null);
+      return;
+    }
+
+    final View flutterView = mFlutterActivity.findViewById(mFlutterViewId);
+    final int initialContentSensitivity = flutterView.getContentSensitivity();
+
+    if (initialContentSensitivity == requestedContentSensitivity) {
+      // Content sensitivity for the requested View already set to requestedContentSensitivity.
+      result.success(null);
+      return;
+    }
+
+    // Set requestedContentSensitivity on the View.
+    flutterView.setContentSensitivity(requestedContentSensitivity);
+
+    // Invalidate the View to force a redraw in order to ensure that it updates to be
+    // obscured/unobscured as expected.
+    flutterView.invalidate();
+
+    result.success(null);
+  }
+
+  /**
+   * Gets content sensitivity level of the Android {@code View} associated with this plugin's {@code
+   * mFlutterViewId}.
+   */
+  @Override
+  public Integer getContentSensitivity() {
+    if (!isSupported()) {
```

Comment:
When you are considering adding an Unknown value also consider if get should return unknown in this situation. 

Even if we dont add unknown it might be worth returning "not sensitive" since we think that is the case on api 35 and prior.

Reply:
Since I'm not implementing that now, going to throw the same error here as `setContentSensitivity` for when `isSupported` is not called first.

Reply:
"Even if we dont add unknown it might be worth returning "not sensitive" since we think that is the case on api 35 and prior." 

Why not make this method safe to call even on earlier versions of android? 
This will especially be important if we back port sensitivity.

Reply:
Agreed. Done.